[Build] libwebrtc not loaded in downlevel Safari GPU process

emw-apple opened the pull request:
#### b3c6569c6a7b269452283e080d9334fb5c49125b
<pre>
[Build] libwebrtc not loaded in downlevel Safari GPU process
<a href="https://rdar.apple.com/125417331">rdar://125417331</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=273231">https://bugs.webkit.org/show_bug.cgi?id=273231</a>

Reviewed by Alexey Proskuryakov.

Set libwebrtc&apos;s DYLIB_INSTALL_NAME_BASE to its installed location inside
of WebCore.framework. Like other WebKit dylibs, its install name
should match the system install path. Downlevel installations provide a
path to the updated frameworks by baking DYLD environment variables into
the binary.

This appears to resolve an issue where the GPU process does not load
libwebrtc in temporary installations of build products. It appears to be
limited to GPU process due to <a href="https://rdar.apple.com/127030110">rdar://127030110</a>.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:

Canonical link: <a href="https://commits.webkit.org/277992@main">https://commits.webkit.org/277992@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/50b6fa1500f54c9ed8173a9a1dbe278f50b831ce

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49153 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28430 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52169 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51951 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45250 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34421 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26016 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40173 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50909 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25992 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42367 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21286 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23447 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43541 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7477 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45376 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44048 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53860 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24231 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20435 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47490 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25509 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42572 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46476 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10813 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26331 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25228 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
> Previously, it was using the temporary staging path it is installed to during the build.

I cannot relate this comment to what the old code did, I don't think that `WK_OVERRIDE_FRAMEWORKS_DIR` is a "during the build" path.

The change looks definitely correct, but I don't understand why we are running into this problem now, and did not run into it a year ago. AFAICT nothing has changed. This seems worth looking into.

emw-apple committed:
[Build] libwebrtc not loaded in downlevel Safari GPU process
rdar://125417331
https://bugs.webkit.org/show_bug.cgi?id=273231

Reviewed by Alexey Proskuryakov.

Set libwebrtc's DYLIB_INSTALL_NAME_BASE to its installed location inside
of WebCore.framework. Like other WebKit dylibs, its install name
should match the system install path. Downlevel installations provide a
path to the updated frameworks by baking DYLD environment variables into
the binary.

This appears to resolve an issue where the GPU process does not load
libwebrtc in temporary installations of build products. It appears to be
limited to GPU process due to rdar://127030110.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:

Canonical link: https://commits.webkit.org/277992@main
